alacritty_terminal/src/term/cell.rs

```diff
@@ -94,25 +95,39 @@ impl Cell {
 
     /// Write a new zerowidth character to this cell.
     #[inline]
-    pub fn push_zerowidth(&mut self, c: char) {
-        self.extra.get_or_insert_with(Default::default).zerowidth.push(c);
-    }
-
-    /// Free all dynamically allocated cell storage.
-    #[inline]
-    pub fn drop_extra(&mut self) {
-        if self.extra.is_some() {
-            self.extra = None;
-        }
+    pub(crate) fn push_zerowidth(&mut self, character: char) {
+        let extra = self.extra.get_or_insert(Arc::new(Default::default()));
+        Arc::make_mut(extra).zerowidth.push(character);
     }
 
     /// Remove all wide char data from a cell.
     #[inline(never)]
-    pub fn clear_wide(&mut self) {
+    pub(crate) fn clear_wide(&mut self) {
```

Comment:
Why are these `pub(crate)`? `CellExtra` is `pub` without crate-specific restrictions, so I'm just curious if there's a reason for this difference?

Reply:
I think it's a left over from Rc. For arc we can make them pub.